db/migrate/20230401123457_create_components.rb

```diff
@@ -0,0 +1,10 @@
+class CreateComponents < ActiveRecord::Migration[6.0]
+  def change
+    create_table :components do |t|
+      t.string :name, null: false
+      t.references :library, null: false, foreign_key: true
```

Comment:
The 'components' table is being created with 'library_id' as a foreign key, but there are no indices on 'library_id'. Consider adding an index to improve query performance.